syntax = "proto3";

package rpc;

message Server {
    string host = 1;
    int32 port = 2;
}

message AddRequest {
    int32 a = 1;
    int32 b = 2;
}

message AddResponse {
    int32 sum = 1;
}

message Request {
    string type = 1;
    string service_name = 2;
    Server server = 3;
    AddRequest add = 4;
}

message Response {
    string type = 1;
    string content = 2;
    repeated Server servers = 3;
    AddResponse add = 4;
}



// service Calculator {
//     rpc add (AddRequest) returns (AddResponse);
// }

// message RegisterRequest {
//     string type = 1;
//     string service_name = 2;
//     string host = 3;
//     int32 port = 4;
// }

// message RegisterResponse {
//   string type = 1;
//   string content = 2;
// }

// message HeartbeatRequest {
//     string type = 1;
//     string service_name = 2;
//     string host = 3;
//     int32 port = 4;
// }

// message HeartbeatResponse {
//     string type = 1;
//     string content = 2;
// }

// message DiscoverRequest {
//     string type = 1;
//     string service_name = 2;
// }

// message DiscoverResponse {
//     string type = 1;
//     repeated Server servers = 2;
// }

// message Server {
//     string host = 1;
//     int32 port = 2;
// }